pkg/virtctl/ssh/ssh.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package ssh
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	portFlag               = "port"
+	wrapLocalSSHFlag       = "localSSH"
+	usernameFlag           = "username"
+	passwordFlag           = "password"
+	privateKeyPathFlag     = "privateKey"
+	privateKeyPasswordFlag = "privateKeyPassword"
+)
+
+var (
+	wrapLocalSSH          bool = false
+	sshPort               int
+	sshUsername           string
+	sshPassword           string
+	sshPrivateKeyPath     string
+	sshPrivateKeyPassword string
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "ssh (VM|VMI)",
+		Short:   "Open a SSH connection to a virtual machine instance.",
+		Example: usage(),
+		Args:    templates.ExactArgs("ssh", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := SSH{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+
+	// Flag is hidden for now, can be removed later
+	cmd.Flags().BoolVar(&wrapLocalSSH, wrapLocalSSHFlag, wrapLocalSSH,
+		fmt.Sprintf("--%s=true: Set this to true to use the SSH command available on your system by using this command as ProxyCommand; If unassigned, this will establish a SSH connection with limited capabilities provided by this client", wrapLocalSSHFlag))
+	cmd.Flags().MarkHidden(wrapLocalSSHFlag)
```

Comment:
I don't think that we should hide this. The agnostic client will never be as powerful as openssh.

Reply:
But at the end we will see in the follow-up PR what to do with it. We can also consider adding a `virtcl config-ssh` command which can for instance add a few nice entries to `.ssh/config` where people can then simply write `ssh vmi/myvm.mynamespace` afterwards and we are not involved anymore at all.